### PR TITLE
Feature/check yaml input

### LINKED ISSUE #95: check yaml input submitted from project/edit page
Do some yaml validation on the project configuration submitted by the project/edit page.

Suggestions:
 * check valid yaml for project config (is currently silently accepted)
 * check valid yaml for env (currently shows the error page)
 * configure the text areas that tabs are converted to 4 spaces
 * maybe do some more checking (valid attributes?, valid plugins?)

Reason:
When I enter my project config, I often create invalid yaml code. Most time it is a tab instead of spaces, or wrong number of spaces.

### PR DESCRIPTION
**Pull request by lugark**
## Contribution type
*Feature - resolves #95 *

## Description of change
- added YAML-validator to build_config + environments TextAreas
- added DataTransformator and implemented YamlTransformator (change from tab to 4 spaces)

### CODE REVIEW

**Comment by corpsee** [comment]
@lugark Thank you for the request. It is the good enhancement. But could you remove your past commits from the branch and rebase the branch on latest `master` branch?

File: src/Form/Input.php [hunk 1]
```diff
@@ -38,6 +39,9 @@ class Input extends Element
      */
     protected $customError = false;
 
+    /** @var DataTransformerInterface */
+    protected $dataTransformator;
```
Comment (corpsee) [inline src/Form/Input.php]: protected $dataTransformator = null;

File: src/Form/Input.php [hunk 2]
```diff
@@ -195,4 +207,20 @@ protected function onPreRender(View &$view)
...
+     */
+    public function getDataTransformator()
+    {
+        return $this->dataTransformator;
+    }
+
+    /**
+     * @param DataTransformerInterface $dataTransformator
+     */
+    public function setDataTransformator($dataTransformator)
```
Comment (corpsee) [inline src/Form/Input.php]: `public function setDataTransformator(DataTransformerInterface $dataTransformator)`

File: composer.json [hunk 3]
```diff
@@ -111,7 +111,8 @@
         "phpunit/php-token-stream":          "~1.4.0",
         "phpdocumentor/reflection-docblock": "~2.0.0",
         "myclabs/deep-copy":                 "~1.7.0",
-        "symfony/polyfill":                  "~1.10.0"
+        "symfony/polyfill":                  "~1.10.0",
+        "symfony/form": "^3.4"
```
[the remark this answers is not in the document]
Comment (lugark) [reply composer.json]: The Idea was to use more of symfony form Features. Easier handling of parameters / sanitazation. Maybe in next Version? 😉
I will refactor it

File: src/Form/Input.php (hunk 2 above)
Comment (lugark) [reply src/Form/Input.php]: Which PSR does the project use? I might have overlooked this information. If it is not written somewhere we should add it

**Comment by lugark** [comment]
Hey - i will refactor the things and create a new request. The history is a bit chaotic and i will start with a fresh pull from your master :)

**Comment by corpsee** [comment]
@ OK. Thank you @lugark!

**Pull request closed without merging**